docs/events/glides/com.mbta.ctd.glides.vehicle_trip_assignments.v1.mdx

```diff
@@ -0,0 +1,184 @@
+- An assignment that includes a `vehicle` but a `null` `tripKey` means that the vehicle is not currently operating a trip.
+- Assignments SHOULD NOT have both fields `null`, but if they do, they can be ignored.
+
+## Updating Assignments
+
+Consumers SHOULD maintain state about which vehicle-trip assignments are active, and update that state when they receive an assignment that refers to a `vehicleId` or `tripKey`.
+
+Vehicles can only be assigned to one trip at a time. A vehicle that was previously assigned to `TRIP A` MAY be assigned directly to `TRIP B` without a separate assignment that assigns its trip to `null` first. Conversely, a trip can only have one assigned vehicle at a time, and doesn't need a previous vehicle to be removed before having a new vehicle assigned to it.
+
+If `VEHICLE 1` was last assigned to `TRIP A`, and a new event assigns `VEHICLE 1` to `TRIP B`, Glides MUST also send another `VehicleTripAssignment` in the same event's batch to assign the `vehicle` for `TRIP A` to `null` or to a different vehicle. Consumers are not expected to deduce that `TRIP A` no longer has a vehicle assigned based on an update to `VEHICLE 1` and `TRIP B` (which doesn't mention `TRIP A`). Glides MUST ensure that after processing all updates in an event's batch, if the last update to `TRIP A` assigned `VEHICLE 1` to it, then `VEHICLE 1` MUST NOT have any other more recent updates.
+
+Conversely, if `VEHICLE 1` was last assigned to `TRIP A`, and a new event assigns `VEHICLE 2` to `TRIP A`, Glides MUST also assign `VEHICLE 1` to another trip or to a `null` trip in the same batch. By the end of processing all of the updates in each batch, if the last update to `VEHICLE 1` assigned it to `TRIP A`, then `TRIP A` MUST NOT have any other more recent updates.
+
+If there are no trains operating on any trips, for example, in the middle of the night after service ends, then for each `vehicleId` and `TripKey` in the feed, the last update to that vehicle or trip will be assigning it to a `null` trip or vehicle.
+
+Notes:
+
+Consumers MUST tolerate vehicles and trips being unassigned (assigned to `null`) even when the consumer didn't know they were previously assigned. This is needed because consumers might not have seen a previous update where they were assigned, as described in the docs on [Short-term storage](index.mdx#short-term-storage): "Clients MUST tolerate receiving updates for events that reference unseen past events". Updates that assign a vehicle or trip to a non-null trip or vehicle are unlikely to cause problems for consumers, though if consumers didn't see all past events, they might not know which trip or vehicle the vehicle or trip is being removed from.
+
+Glides MAY publish an event that assigns a vehicle to a trip even if the last update for that vehicle and trip assigned them to each other. Glides might do this to update the `carNumbers` associated with a `vehicleId`.
```

Comment:
**suggestion:** I understand the rationale for including car numbers as a sort of reference for humans looking at the data, but I worry that this sort of implies that Glides would update the consist for a vehicle by publishing new events. I think it should be specified that Glides won't necessarily do this.

Reply:
How could the spec avoid making that implication? Maybe in the `Vehicle` section, say something like "for context only, it isn't meant to be an authoritative source for car numbers, use GTFS-RT from RTR instead if you want to know what cars are associated with a vehicleId?"

Or maybe it'd be easier to just lean in to it and guarantee that we send updates for car numbers if they change? The car numbers come from GTFS-RT, so should be just as accurate as other sources, as long as Glides promises to keep them up to date.

Reply:
> Or maybe it'd be easier to just lean in to it and guarantee that we send updates for car numbers if they change?

For what it's worth, I would strongly prefer we not go this direction, and instead just make it explicit that the only time one of these messages is guaranteed to be generated is when the trip assignment state changes.

Reply:
Where is Glides getting vehicle information in the current version? This has moved around a bit, and I'm not sure where we've settled on at the moment.

If the plan is to get it from an RTR feed rather than reading the trike feed, I don't think there's too much value in car numbers. It does save a human a round trip, but RTR will already have logged them, and the vehicle ID will be consistent between both, so it's really only for if the human happens to be reading the raw events.

On the other hand, if Glides is reading Trike directly, this could be a handy sanity check.

Overall, I think I lean on the side of leaving consist out. I don't like the idea of having data in the event that may go stale, if it's there I kind of feel like we should keep it up-to-date, but issuing a new TripAssignment for consist doesn't make sense semantically.

If we did keep it, I think I'd argue for broadening the scope, where this is something like a VehicleUpdate rather than a TripAssignment each event be the full state of a vehicle (it's assigned to + consist + any other information we wanted), where any change to vehicle information could trigger it, but I doubt rethinking this PR in that direction is appealing to anyone.

Reply:
It will be some amount of effort to look up car numbers when debugging, but that data does exist in a few sources. The cost from the complexity will be worse than the extra debugging effort, so I'll remove it.

I'll also un-nest vehicle_id into the top level, removing the vehicle object.